### PR TITLE
feat: Add auto-assign on reply based on user preference

### LINKED ISSUE #1008: "Auto-assign replies to self" setting
### What

Setting per user to automatically assign tickets to yourself when you reply (if unassigned?)

### Why

Several people do this manually which takes time.

### PR DESCRIPTION
**Pull request by praveenkalady**
### What's New
- New user preference: **"Auto-assign on reply"** in Settings > Preferences
- Automatically assigns unassigned conversations to users when they reply
- Streamlines workflow by combining reply + assignment into single action

### Key Features
- ✅ Optional setting (disabled by default)
- ✅ Only affects unassigned conversations
- ✅ Works with existing reply functionality
- ✅ Preserves existing assignments

### Technical Changes
- Added `autoAssignOnReply` to user preferences schema
- Enhanced reply logic with auto-assignment check
- Added settings UI with toggle and saving indicator
- Full e2e test coverage for all scenarios

#### Before (Manual Assignment)

https://github.com/user-attachments/assets/b28bd50b-3d0e-4b00-af8d-a22a4934f7ea

#### After (Auto-Assign Enabled)

https://github.com/user-attachments/assets/03f24945-ed56-4267-9179-a10bfa6763ee

#### Testing

- `preferences.spec.ts for auto assign settings`
Screenshot 2025-09-04 at 4 18 52 AM

- `conversationActions.spec.ts for auto assign feature test`
Screenshot 2025-09-04 at 4 19 42 AM

### AI disclosure 

AI used to plan and assist (Claude opus 4.1).

### CODE REVIEW

File: tests/e2e/conversations/conversationActions.spec.ts [hunk 1]
```diff
@@ -1,39 +1,20 @@
...
-    .select({ changes: conversationEvents.changes })
-    .from(conversationEvents)
-    .where(eq(conversationEvents.conversationId, conversationId))
-    .orderBy(desc(conversationEvents.createdAt))
-    .limit(1);
-  if (event && event.changes && event.changes.status) {
-    return event.changes.status;
-  }
-  return "unknown";
-}
```
Comment (praveenkalady) [inline tests/e2e/conversations/conversationActions.spec.ts:17]: Moved helper functions to test-helpers.ts for improved maintainability.

**Comment by praveenkalady** [comment]
@binary-koan Could you please review? The PR is ready, well tested, and covers all edge cases. I have also done a self-review.

**Review by binary-koan — APPROVED** [review]
Spent some time simplifying the e2e tests, but still picking this one to merge as it's the only one that includes tests for the actual conversation view.

LGTM, thanks!
